lib/assetreqs/assetreqs.go

```diff
@@ -55,7 +67,53 @@ func New(token string, assetName string, orderNum string, cadenceName string, ca
 		fPath: filePath,
 		fName: fileName,
 		oFmt:  outputFormat,
+	}, nil
+}
+
+func ordNumIsValidFmt(orderNum string) (v bool, err error) {
+	// Note: because of no look-ahead operator (?) in go regexp, this takes multiple evaluations.
+	// Make sure that the given order number has at least one number and at least one letter, contains only digits and
+	// letters, and is 6 characters long.
+
+	// Test length = 6.
+	if len(orderNum) != 6 {
+		return false, nil
+	}
+
+	// Length is correct - make sure it doesn't contain something other than letters or digits.
+	invalid, err := hasNonLetterOrNonDigit(orderNum)
+	if err != nil {
+		return false, err
+	}
+	if invalid {
+		// contains something other than letters and digits - invalid
+		return false, err
 	}
+
+	// Make sure it contains at least one letter.
+	v, err = hasLetter(orderNum)
```

Comment:
I don't believe there's a order number requirement that it must have a letter in the order number. I know most (if not all) do currently in production, but I see orders in our test env that have all numbers with no letters.

Reply:
this is intended for production environment and viya 4 orders only. for the foreseeable future, those orders will always contain a letter.

Reply:
hmm, okay, eventually then if this API lives for a long time then this part of the validation will need to be removed. That said, it's most likely this won't happen for a long time.